src/plugins/explore/public/services/tab_registry/tab_registry_service.ts

```diff
@@ -34,15 +34,17 @@ export interface TabDefinition {
   // Transform query string for cache key generation
   prepareQuery?: (query: Query) => string;
 
-  // Optional results processor for raw results
-  resultsProcessor?: (rawResults: any, indexPattern: any, includeHistogram?: boolean) => any;
+  /**
+   * @experimental Callback for errors from query
+   * Used for handling patterns queries which fail on older versions.
+   * Will be removed when a method of finding dataset query engine version is implemented
+   *
+   * @returns A custom QueryExecutionStatus as well as an error, or undefined if standard behavior is desired
+   */
+  handleQueryError?: (error: any, cacheKey: string) => boolean;
```

Comment:
will this cause issues?

Reply:
some potential issues (from our discussion but for everyone else):
- security related issues (the 'correct' user flow means sending out a call that will return with a 400)
- potential race conditions (in between the time waiting for calls, regex state being changed, strange UI results)

Reply:
a potential future solution that could be implemented which wouldn't have these issues (from your suggestion):
server side handling of this flow.
1) checking if the query server can handle 'brain' pattern queries, by sending out that query
2) intercepting an error, checking to see if it matches the criteria of the server being too old
3) sending out the older patterns query, and marking in the result that it is so
This would solve the two issues above.